### DISCUSSION TITLE
Poll: Better tagline for Ibis

### DISCUSSION THREAD

**Discussion by NickCrews** [ibis-project/ibis#7137] [Proposals]
Discussion in https://github.com/ibis-project/ibis/discussions/3701 talked about how the tagline for ibis (currently "the portable Python dataframe library") could do with some workshopping. I wanted to open this up to the broader community because I think it is really important to get this tagline right, because this is what everyone will use whenever they try to describe Ibis to their coworkers, in blog posts, on reddit, etc.

First, I thought I'd list out some of the keywords that might belong (based off https://ibis-project.org/why, the advertised merits of [duckdb](https://duckdb.org/), and my experience):

- dataframe
- sql
- portable
- scalable / powerful
- extensible
- developer-friendly
- full-featured / feature-rich
- easy
- simple
- free

I personally rank the scalability over portability, so I would vote for "The SQL-fueled python dataframe library". Fueled trying to imply "powerful". 

Did I miss a keyword you think is important? What would your tagline be? Once we have some candidates then perhaps we vote?

**Comment by p-a-a-a-trick** [comment]
Thanks for starting this, Nick!

I don't have one off the top of my head that I'd commit to, but IMO SQL seems a bit limiting - it feels like it's excluding backends like pyspark, dask, polars (or any other non-SQL backend, current or future); it kinda makes it seem limited to traditional databases.

**Reply by NickCrews** [reply]
Hmmmm, I see. Technically, all of those are actually queried using SQL, but your point still holds. I see SQL and I don't think Polars.

Maybe like "Python's [feature-complete, mature, de-facto, etc] dataframe interface to 15+ [backends, powerful backends, powerful compute engines]"?

I like the "interface", as it implies portability, and that the power of the engines is just passed through. It's more precise than "library" too.

"Python's" serves the purpose of the original leading "the", implying de-facto, the one and only. Plus it cuts down on words.

**Reply by lostmygithubaccount** [reply]
continuing on that thread -- what about "Python's portable dataframe library: the standard interface to 15+ data platforms"

**Reply by NickCrews** [reply]
I like platforms. Maybe a little more succinct "Python's portable dataframe interface to 15+ data platforms", do you think the "library" adds something that "interface" doesn't cover?

**Reply by cpcloud** [reply]
Not trolling, but we're at 18 now :)

**Reply by jcrist** [reply]
I'm not really a fan of a tagline that we have to increment periodically. IMO ideally the tagline is something memorable that can stay constant, info on how many backends we support can occur elsewhere in the docs.

**Reply by lostmygithubaccount** [reply]
but 15 is a round number!

@ianmcook's point on "library" (versus "interface") has been around familiarity for more casual users -- a library implies something they can pip install and get using, whereas an interface may be associated w/ a UI or API. I'm not super strongly opinionated on using library vs interface in the tagline...we can run an A/B test one day 🙃 

"data platform" is a good industry word to stay neutral and avoid pedantic discussions on database vs data warehouse vs data lake vs lakehouse vs Python dataframe vs Rust dataframe w/ Python wrapper...

btw I have the # of backends and list of them in a dashboard here: https://ibis-analytics.streamlit.app/ it will update automatically as new ones are added

> I'm not really a fan of a tagline that we have to increment periodically. IMO ideally the tagline is something memorable that can stay constant, info on how many backends we support can occur elsewhere in the docs.

I also agree with this -- I'd prefer a shorter "Python's portable dataframe library/interface", then in prose on whatever page(s) we can add "that supports 18+ data platforms, providing a unified interface..."

**Reply by ianmcook** [reply]
I think the possessive "Python's" carries a potentially misleading connotation that Ibis is part of core Python or the standard library

**Reply by NickCrews** [reply]
> "Python's" carries a potentially misleading connotation

Yeah that's fair, probably should avoid that. I still like the "The" prefix though, because I think it is fair to claim ibis as the only production-ready project to do this.

> whereas an interface may be associated w/ a UI or API

OK, I see that drawback too. library works

Removing the 15+ gives us space for another adjective! And we could emphasize that we probably already support __**your**__ backend: "The portable, scalable Python dataframe library for all your data platforms"

**Comment by ianmcook** [comment]
What exactly would you say are the shortcomings of the current tagline ("the portable Python dataframe library")?

IMO changing the tagline again seems premature unless we can point to specific problems with the current tagline that are decisively resolved by a new tagline.

**Reply by NickCrews** [reply]
I moved to ibis from pandas because it gave be a familiar and composable dataframe API, but was more powerful/scalable using duckdb. The current tagline would not have sold me. I had to dive deeper before I realized that Ibis solved my problem. The current tagline is accurate, but doesn't highlight the features that I think are most convincing. But that might just be because of my experience, other people might care about portability more.

**Reply by ianmcook** [reply]
What if we used the word "universal" instead of "portable"?

>The universal Python dataframe library

That's more aspirational than accurate, and I'm not sure we're ready to be that bold... but would that tagline have sold you?

**Reply by NickCrews** [reply]
I think I prefer "portable" but "universal" wouldn't make me cry :)

> would that tagline have sold you?

No, because it still doesn't evoke "scalable/powerful" :(

**Reply by ianmcook** [reply]
Thanks. I think we feel some  hesitation to attach adjectives like "scalable" or "powerful" directly to Ibis, because those benefits come from the backends, not from Ibis itself

**Reply by NickCrews** [reply]
I see that, more so with powerful. I think scalable is a fair claim though (and I'm not on your marketing team so that means you have permission 😉 ). The second sentence at https://ibis-project.org/why is "If you’ve had issues with scaling data transformation code in Python, ...", and that scalability is coming directly from Ibis
